src/Tasks/Microsoft.NET.Build.Tasks/GetPackagesToPrune.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.ComponentDetection.Detectors.NuGet;
+using NuGet.Frameworks;
+using NuGet.Versioning;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public class GetPackagesToPrune : TaskBase
+    {
+        [Required]
+        public string TargetFrameworkIdentifier { get; set; }
+
+        [Required]
+        public string TargetFrameworkVersion { get; set; }
+
+        [Required]
+        public ITaskItem[] FrameworkReferences { get; set; }
+
+        [Output]
+        public ITaskItem[] PackagesToPrune { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            var nugetFramework = new NuGetFramework(TargetFrameworkIdentifier, Version.Parse(TargetFrameworkVersion));
+
+            Dictionary<string, NuGetVersion> packagesToPrune = new();
+
+            var frameworkPackages = FrameworkPackages.GetFrameworkPackages(nugetFramework, FrameworkReferences.Select(fr => fr.ItemSpec).ToArray())
+                .SelectMany(packages => packages);
+
+            foreach (var kvp in frameworkPackages)
+            {
+                if (packagesToPrune.TryGetValue(kvp.Key, out NuGetVersion existingVersion))
+                {
+                    if (kvp.Value > existingVersion)
+                    {
+                        packagesToPrune[kvp.Key] = kvp.Value;
```

Comment:
Did we have packages overlapping between different framework references?

Reply:
I got a warning about multiple `PrunePackageReference` items when I was testing this, so I added the deduplication code.  It looks like it happens when referencing multiple shared frameworks.  I didn't investigate to see where the duplication was coming from, but I do see different version numbers too.

Reply:
I see, WindowsDesktop does overlap with AspNetCore.